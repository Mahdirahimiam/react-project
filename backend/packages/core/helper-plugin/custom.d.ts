export {};

declare global {
  interface Window {
    strapi: {
      backendURL: string;
      telemetryDisabled: boolean;
      projectType: 'Enterprise' | 'Community';
    };
  }
}

import { StrapiTheme } from '@strapi/design-system';

declare module 'styled-components' {
  // eslint-disable-next-line @typescript-eslint/no-empty-interface
  export interface DefaultTheme extends StrapiTheme {}
}
