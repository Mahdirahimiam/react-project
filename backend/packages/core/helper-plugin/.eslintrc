{
  "root": true,
  "extends": ["custom/front/typescript", "plugin:storybook/recommended"],
  "rules": {
    "import/no-unresolved": [
      "error",
      {
        // Eslint does not detect the package.json exports field
        // and reports broken imports for the following imports
        "ignore": ["@strapi/design-system/v2"]
      }
    ]
  },
  "overrides": [
    {
      "files": ["./jest.config.front.js", "./webpack.config.js"],
      "env": {
        "node": true
      },
      "rules": {
        "@typescript-eslint/no-var-requires": "off"
      }
    },
    {
      "files": ["./tests/*"],
      "env": {
        "jest": true
      }
    }
  ]
}
