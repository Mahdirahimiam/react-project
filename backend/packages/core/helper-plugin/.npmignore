############################
# dts
############################

dist/**/*.d.ts.map