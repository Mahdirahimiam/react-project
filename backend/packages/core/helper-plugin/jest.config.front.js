module.exports = {
  preset: '../../../jest-preset.front.js',
  collectCoverageFrom: ['<rootDir>/packages/core/helper-plugin/src/**/*.js'],
  displayName: 'Helper plugin',
  moduleNameMapper: {
    '^@tests/(.*)$': '<rootDir>/tests/$1',
  },
  setupFilesAfterEnv: ['./tests/setup.ts'],
};
