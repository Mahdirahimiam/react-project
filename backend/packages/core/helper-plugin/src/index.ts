/* -------------------------------------------------------------------------------------------------
 * Components
 * -----------------------------------------------------------------------------------------------*/

export * from './components/AnErrorOccurred';
export * from './components/CheckPagePermissions';
export * from './components/CheckPermissions';
export * from './components/ConfirmDialog';
export * from './components/ContentBox';
export * from './components/DateTimePicker';
export * from './components/DynamicTable';
export * from './components/Table';
export * from './components/EmptyStateLayout';
export * from './components/EmptyBodyTable';
export * from './components/FilterListURLQuery';
export * from './components/FilterPopoverURLQuery';
export * from './components/Form';
export * from './components/GenericInput';
export * from './components/InjectionZone';
export * from './components/Link';
export * from './components/LinkButton';
export * from './components/LoadingIndicatorPage';
export * from './components/NoContent';
export * from './components/NoMedia';
export * from './components/NoPermissions';
export * from './components/NotAllowedInput';
export * from './components/PageSizeURLQuery';
export * from './components/PaginationURLQuery';
export * from './components/ReactSelect';
export * from './components/RelativeTime';
export * from './components/SearchURLQuery';
export * from './components/SettingsPageTitle';
export * from './components/Status';

/* -------------------------------------------------------------------------------------------------
 * Content Manager
 * -----------------------------------------------------------------------------------------------*/

export {
  useCMEditViewDataManager,
  ContentManagerEditViewDataManagerContext,
  type CMEditViewDataManagerContextValue,
  type ContentType,
} from './content-manager/CMEditViewDataManager';
export * from './content-manager/utils/getAttributeInfos';

/* -------------------------------------------------------------------------------------------------
 * Features
 * -----------------------------------------------------------------------------------------------*/

export * from './features/AppInfo';
export * from './features/AutoReloadOverlayBlocker';
export * from './features/CustomFields';
export * from './features/GuidedTour';
export * from './features/Library';
export * from './features/Notifications';
export * from './features/OverlayBlocker';
export * from './features/RBAC';
export * from './features/StrapiApp';
export * from './features/Tracking';

/* -------------------------------------------------------------------------------------------------
 * Hooks
 * -----------------------------------------------------------------------------------------------*/

export * from './hooks/useAPIErrorHandler';
export * from './hooks/useCallbackRef';
export * from './hooks/useClipboard';
export * from './hooks/useCollator';
export * from './hooks/useFetchClient';
export * from './hooks/useFilter';
export * from './hooks/useFocusInputField';
export * from './hooks/useFocusWhenNavigate';
export * from './hooks/useLockScroll';
export * from './hooks/usePersistentState';
export * from './hooks/useQuery';
export * from './hooks/useQueryParams';
export * from './hooks/useRBAC';
export * from './hooks/useSelectionState';

/* -------------------------------------------------------------------------------------------------
 * Icons
 * -----------------------------------------------------------------------------------------------*/

export * from './icons/RemoveRoundedButton';
export * from './icons/SortIcon';

/* -------------------------------------------------------------------------------------------------
 * Utils
 * -----------------------------------------------------------------------------------------------*/

export * from './content-manager/utils/contentManagementUtilRemoveFieldsFromData';
export * from './content-manager/utils/formatContentTypeData';
export * from './utils/auth';
export * from './utils/awaitToJs';
export * from './utils/difference';
export * from './utils/getAPIInnerErrors';
export * from './utils/getFetchClient';
export * from './utils/getFileExtension';
export * from './utils/getYupInnerErrors';
export * from './utils/hasPermissions';
export * from './utils/normalizeAPIError';
export * from './utils/prefixFileUrlWithBackendUrl';
export * from './utils/prefixPluginTranslations';
export * from './utils/pxToRem';
export * from './utils/request';
export * from './utils/setHexOpacity';
export * from './utils/stopPropagation';
export * from './utils/translatedErrors';
export * from './utils/wrapAxiosInstance';

/* -------------------------------------------------------------------------------------------------
 * Types
 * -----------------------------------------------------------------------------------------------*/

export type { TranslationMessage, FilterData } from './types';
