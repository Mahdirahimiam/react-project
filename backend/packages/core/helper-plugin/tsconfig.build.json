{
  "extends": "./tsconfig.json",
  "exclude": ["**/*.test.tsx", "**/*.test.ts"],
  "compilerOptions": {
    "declarationMap": true,
    "outDir": "./dist"
  }
}
