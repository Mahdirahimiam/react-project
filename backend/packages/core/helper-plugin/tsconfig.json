{
  "extends": "tsconfig/client.json",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@tests/*": ["./tests/*"]
    }
  },
  "include": ["src", "custom.d.ts"]
}
