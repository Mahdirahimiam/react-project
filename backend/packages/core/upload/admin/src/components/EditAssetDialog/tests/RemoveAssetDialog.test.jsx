import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { NotificationsProvider } from '@strapi/helper-plugin';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { QueryClient, QueryClientProvider } from 'react-query';

import en from '../../../translations/en.json';
import { RemoveAssetDialog } from '../RemoveAssetDialog';

jest.mock('../../../utils/deleteRequest', () => ({
  ...jest.requireActual('../../../utils/deleteRequest'),
  deleteRequest: jest.fn().mockResolvedValue({ id: 1 }),
}));

const messageForPlugin = Object.keys(en).reduce((acc, curr) => {
  acc[curr] = `upload.${en[curr]}`;

  return acc;
}, {});

const asset = {
  id: 8,
  name: 'Screenshot 2.png',
  alternativeText: null,
  caption: null,
  width: 1476,
  height: 780,
  formats: {
    thumbnail: {
      name: 'thumbnail_Screenshot 2.png',
      hash: 'thumbnail_Screenshot_2_5d4a574d61',
      ext: '.png',
      mime: 'image/png',
      width: 245,
      height: 129,
      size: 10.7,
      path: null,
      url: '/uploads/thumbnail_Screenshot_2_5d4a574d61.png',
    },
    large: {
      name: 'large_Screenshot 2.png',
      hash: 'large_Screenshot_2_5d4a574d61',
      ext: '.png',
      mime: 'image/png',
      width: 1000,
      height: 528,
      size: 97.1,
      path: null,
      url: '/uploads/large_Screenshot_2_5d4a574d61.png',
    },
    medium: {
      name: 'medium_Screenshot 2.png',
      hash: 'medium_Screenshot_2_5d4a574d61',
      ext: '.png',
      mime: 'image/png',
      width: 750,
      height: 396,
      size: 58.7,
      path: null,
      url: '/uploads/medium_Screenshot_2_5d4a574d61.png',
    },
    small: {
      name: 'small_Screenshot 2.png',
      hash: 'small_Screenshot_2_5d4a574d61',
      ext: '.png',
      mime: 'image/png',
      width: 500,
      height: 264,
      size: 31.06,
      path: null,
      url: '/uploads/small_Screenshot_2_5d4a574d61.png',
    },
  },
  hash: 'Screenshot_2_5d4a574d61',
  ext: '.png',
  mime: 'image/png',
  size: 102.01,
  url: '/uploads/Screenshot_2_5d4a574d61.png',
  previewUrl: null,
  provider: 'local',
  provider_metadata: null,
  createdAt: '2021-10-04T09:42:31.670Z',
  updatedAt: '2021-10-04T09:42:31.670Z',
};

const renderCompo = (handleCloseSpy = jest.fn()) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        refetchOnWindowFocus: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={lightTheme}>
        <IntlProvider locale="en" messages={messageForPlugin} defaultLocale="en">
          <NotificationsProvider>
            <RemoveAssetDialog onClose={handleCloseSpy} asset={asset} />
          </NotificationsProvider>
        </IntlProvider>
      </ThemeProvider>
    </QueryClientProvider>,
    { container: document.getElementById('app') }
  );
};

describe('RemoveAssetDialog', () => {
  it('snapshots the component', () => {
    renderCompo();

    expect(document.body).toMatchSnapshot();
  });

  it('closes the dialog when pressing cancel', () => {
    const handleCloseSpy = jest.fn();
    renderCompo(handleCloseSpy);

    fireEvent.click(screen.getByText('Cancel'));
    expect(handleCloseSpy).toHaveBeenCalled();
  });

  describe('remove asset', () => {
    it('closes the dialog when everything is going okay when removing', async () => {
      const handleCloseSpy = jest.fn();
      renderCompo(handleCloseSpy);

      fireEvent.click(screen.getByText('Confirm'));

      await waitFor(() => expect(handleCloseSpy).toHaveBeenCalled());

      expect(screen.getByText(/Elements have been successfully deleted/)).toBeInTheDocument();
    });
  });
});
