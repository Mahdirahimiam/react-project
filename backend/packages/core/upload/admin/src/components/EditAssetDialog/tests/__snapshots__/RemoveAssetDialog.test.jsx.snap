// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveAssetDialog snapshots the component 1`] = `
.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c11 {
  font-weight: 600;
  font-size: 1.125rem;
  line-height: 1.22;
  color: #32324d;
}

.c15 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #32324d;
}

.c22 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #ffffff;
}

.c0 {
  margin-left: -250px;
  position: fixed;
  left: 50%;
  top: 2.875rem;
  z-index: 10;
  width: 31.25rem;
}

.c3 {
  padding: 40px;
  position: fixed;
  z-index: 4;
}

.c5 {
  background: #ffffff;
  border-radius: 4px;
  box-shadow: 0px 2px 15px rgba(33,33,52,0.1);
}

.c7 {
  padding: 24px;
}

.c12 {
  padding-top: 40px;
  padding-right: 24px;
  padding-bottom: 40px;
  padding-left: 24px;
}

.c13 {
  padding-bottom: 8px;
}

.c16 {
  padding: 16px;
}

.c19 {
  background: #4945ff;
  padding: 8px;
  padding-right: 16px;
  padding-left: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c1 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 8px;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c18 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c20 {
  position: relative;
  outline: none;
}

.c20 > svg {
  height: 12px;
  width: 12px;
}

.c20 > svg > g,
.c20 > svg path {
  fill: #ffffff;
}

.c20[aria-disabled='true'] {
  pointer-events: none;
}

.c20:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c20:focus-visible {
  outline: none;
}

.c20:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c21 {
  height: 2rem;
  border: 1px solid #dcdce4;
  background: #ffffff;
}

.c21 svg {
  height: 0.75rem;
  width: auto;
}

.c21[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c21[aria-disabled='true'] .c10 {
  color: #666687;
}

.c21[aria-disabled='true'] svg > g,
.c21[aria-disabled='true'] svg path {
  fill: #666687;
}

.c21[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c21[aria-disabled='true']:active .c10 {
  color: #666687;
}

.c21[aria-disabled='true']:active svg > g,
.c21[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c21:hover {
  background-color: #f6f6f9;
}

.c21:active {
  background-color: #eaeaef;
}

.c21 .c10 {
  color: #32324d;
}

.c21 svg > g,
.c21 svg path {
  fill: #32324d;
}

.c23 {
  height: 2rem;
  border: 1px solid #f5c0b8;
  background: #fcecea;
}

.c23 svg {
  height: 0.75rem;
  width: auto;
}

.c23[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c23[aria-disabled='true'] .c10 {
  color: #666687;
}

.c23[aria-disabled='true'] svg > g,
.c23[aria-disabled='true'] svg path {
  fill: #666687;
}

.c23[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c23[aria-disabled='true']:active .c10 {
  color: #666687;
}

.c23[aria-disabled='true']:active svg > g,
.c23[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c23:hover {
  background-color: #ffffff;
}

.c23:active {
  background-color: #ffffff;
  border: 1px solid #d02b20;
}

.c23:active .c10 {
  color: #d02b20;
}

.c23:active svg > g,
.c23:active svg path {
  fill: #d02b20;
}

.c23 .c10 {
  color: #b72b1a;
}

.c23 svg > g,
.c23 svg path {
  fill: #b72b1a;
}

.c4 {
  inset: 0;
  background: #32324d33;
}

.c6 {
  max-width: 25.75rem;
  margin: 0 auto;
  overflow: hidden;
  margin-top: 10%;
}

.c9 {
  border-bottom: 1px solid #eaeaef;
}

.c14 svg {
  width: 24px;
  height: 24px;
}

.c14 svg path {
  fill: #d02b20;
}

.c17 {
  border-top: 1px solid #eaeaef;
}

.c17 button {
  width: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<body
  class="lock-body-scroll"
>
  <div>
    <div
      class="c0 c1"
    />
    <div
      class="c2"
    >
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-log"
        role="log"
      />
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-status"
        role="status"
      />
      <p
        aria-live="assertive"
        aria-relevant="all"
        id="live-region-alert"
        role="alert"
      />
    </div>
  </div>
  <div
    class=""
  >
    <div
      class="c3 c4"
    >
      <div>
        <div>
          <div
            aria-labelledby="confirmation-label"
            aria-modal="true"
            class="c5 c6"
            role="dialog"
          >
            <div
              class="c7 c8 c9"
            >
              <h2
                class="c10 c11"
                id="confirmation-label"
              >
                Confirmation
              </h2>
            </div>
            <div
              class=""
            >
              <div
                class=""
                id="confirm-description"
              >
                <div
                  class="c12"
                >
                  <div
                    class="c13 c14"
                  >
                    <div
                      class="c8"
                    >
                      <svg
                        fill="none"
                        height="1rem"
                        viewBox="0 0 24 24"
                        width="1rem"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 0C5.383 0 0 5.383 0 12s5.383 12 12 12 12-5.383 12-12S18.617 0 12 0Zm1.154 18.456h-2.308V16.15h2.308v2.307Zm-.23-3.687h-1.847l-.346-9.23h2.538l-.346 9.23Z"
                          fill="#212134"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="c1"
                  >
                    <div
                      class="c8"
                    >
                      <span
                        class="c10 c15"
                      >
                        Are you sure you want to delete this?
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="c16 c17"
              >
                <div
                  class="c18"
                >
                  <button
                    aria-disabled="false"
                    class="c19 c18 c20 c21"
                    type="button"
                  >
                    <span
                      class="c10 c22"
                    >
                      Cancel
                    </span>
                  </button>
                  <button
                    aria-disabled="false"
                    class="c19 c18 c20 c23"
                    id="confirm-delete"
                    type="button"
                  >
                    <div
                      aria-hidden="true"
                      class=""
                    >
                      <svg
                        fill="none"
                        height="1rem"
                        viewBox="0 0 24 24"
                        width="1rem"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3.236 6.149a.2.2 0 0 0-.197.233L6 24h12l2.96-17.618a.2.2 0 0 0-.196-.233H3.236ZM21.8 1.983c.11 0 .2.09.2.2v1.584a.2.2 0 0 1-.2.2H2.2a.2.2 0 0 1-.2-.2V2.183c0-.11.09-.2.2-.2h5.511c.9 0 1.631-1.09 1.631-1.983h5.316c0 .894.73 1.983 1.631 1.983H21.8Z"
                          fill="#32324D"
                        />
                      </svg>
                    </div>
                    <span
                      class="c10 c22"
                    >
                      Confirm
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
